### PR TITLE
WT-9860 Debug dumps include page deletion information for fast truncate

### PR DESCRIPTION
**Pull request by GWlodarek**
[Patch build](https://spruce.mongodb.com/version/635ef16ad6d80a5ef41434f0/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

### CODE REVIEW

File: src/btree/bt_debug.c [hunk 1]
```diff
@@ -1638,6 +1652,12 @@ __debug_ref(WT_DBG *ds, WT_REF *ref)
     if (__wt_ref_addr_copy(session, ref, &addr) && !WT_TIME_AGGREGATE_IS_EMPTY(&addr.ta))
         WT_RET(ds->f(ds, ", %s, %s", __wt_time_aggregate_to_string(&addr.ta, time_string),
           __wt_addr_string(session, addr.addr, addr.size, ds->t1)));
+    if (ref->state == WT_REF_DELETED && ref->page_del != NULL) {
```
Comment (etienneptl) [inline src/btree/bt_debug.c:1655]: @kommiharibabu, do we still need to check for the `state` here? With WT-9987, we prove that the information contained by `page_del` could be interesting when the page is in memory.

File: src/btree/bt_debug.c [hunk 2]
```diff
@@ -559,6 +560,18 @@ __debug_cell_int(WT_DBG *ds, const WT_PAGE_HEADER *dsk, WT_CELL_UNPACK_ADDR *unp
         break;
     }
 
+    /* Dump the deleted pages transaction ID, commit timestamp, and durable timestamp. */
+    switch (unpack->raw) {
```
Comment (etienneptl) [inline src/btree/bt_debug.c:564]: Can we reuse the `switch` just below?

File: src/btree/bt_debug.c (hunk 1 above)
Comment (kommiharibabu) [reply src/btree/bt_debug.c:1655]: Yes, we don't need to check for `WT_REF_DELETED`. Whenever the `page_del` is not NULL, dump the information.

File: src/btree/bt_debug.c (hunk 2 above)
Comment (GWlodarek) [reply src/btree/bt_debug.c:564]: Sure thing

File: src/btree/bt_debug.c (hunk 1 above)
Comment (GWlodarek) [reply src/btree/bt_debug.c]: I removed the check

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by etienneptl — APPROVED** [review]
LGTM - I suggest creating a function to avoid code duplication.
The code in `__debug_ref` was tested but not the one in `__debug_cell_int`, make sure to test it.

**Pull request merged**